sklearn/covariance/_shrunk_covariance.py

```diff
@@ -428,7 +429,8 @@ def fit(self, X, y=None):
         """
         # Not calling the parent object to fit, to avoid computing the
         # covariance matrix (and potentially the precision)
-        X = self._validate_data(X)
+        with config_context(assume_finite=True):
+            X = self._validate_data(X)
```

Comment:
I would rather do the opposite: use `assume_finite=True` around the call to the `ledoit_wolf` function at line 438 below and keep the `_validate_date` the way it is.

Reply:
@ogrisel Yep you're right!